kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,895 @@
+     * @details Some solvers may require a minimum degree of knowledge of the structure of the matrix.
+     * For example, when solving a mixed u-p problem, it is important to identify the row associated with v and p.
+     * Another example is the automatic prescription of rotation null-space for smoothed-aggregation solvers,
+     * which require knowledge of the spatial position of the nodes associated with a given degree of freedom (DOF).
+     * This function provides the opportunity to provide such data if needed.
+     * @param rA The sparse matrix.
+     * @param rX The solution vector.
+     * @param rB The right-hand side vector.
+     * @param rDoFSet The set of degrees of freedom.
+     * @param rModelPart The model part.
+     */
+    void ProvideAdditionalData(
+        SparseMatrixType& rA,
+        VectorType& rX,
+        VectorType& rB,
+        typename ModelPart::DofsArrayType& rDoFSet,
+        ModelPart& rModelPart
+        ) override
+    {
+        GetCurrentSolver()->ProvideAdditionalData(rA, rX, rB, rDoFSet, rModelPart);
+    }
+
+    /**
+     * @brief Add a linear solver to the collection.
+     * @details This function adds a linear solver to the collection without extending parameters.
+     * @param pSolver Pointer to the linear solver to be added.
+     */
+    void AddSolver(LinearSolverPointer pSolver)
+    {
+        // Increase the solvers vector
+        mSolvers.push_back(pSolver);
+
+        // Extend the parameters
+        FillParametersFromSolver(pSolver);
+    }
+
+    /**
+     * @brief Add a linear solver to the collection with additional parameters.
+     * @details This function adds a linear solver to the collection and extends the parameters.
+     * @param ThisParameters Parameters associated with the linear solver.
+     */
+    void AddSolver(const Parameters ThisParameters)
+    {
+        // Create the solver
+        auto p_solver = ConstructLinearSolverFromSettings(ThisParameters);
+        mSolvers.push_back(p_solver);
+
+        // Add the new solver parameters to the collection
+        mParameters["solvers"].Append(ThisParameters);
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    /**
+     * @brief Get the Solvers list.
+     * @return const std::vector<LinearSolverPointer>& Reference to the vector of solvers.
+     */
+    const std::vector<LinearSolverPointer>& GetSolvers() const
+    {
+        return mSolvers;
+    }
+
+    /**
+     * @brief Set the Solvers list.
+     * @param rSolvers A vector of LinearSolverPointer to set.
+     */
+    void SetSolvers(const std::vector<LinearSolverPointer>& rSolvers)
+    {
+        // Assign the solvers
+        mSolvers = rSolvers;
+
+        // Remove solvers and add again
+        mParameters.RemoveValue("solvers");
+        mParameters.AddEmptyArray("solvers");
+
+        // Fill the parameters with the solvers
+        for (auto& p_solver : mSolvers) {
+            FillParametersFromSolver(p_solver);
+        }
+    }
```

Comment:
are all those methods really needed except for testing?

The behavior is tricky to get right, I would prefer to add them only if needed. Can you outline a usecase please?

Reply:
I think it could be interesting to have those methods, but yes, for standard workflow will not be used, but may be someone can use it for some benchmarkign or test case and may be interesting.